reputation-commitments/src/tests.rs

```diff
@@ -0,0 +1,399 @@
+
+			// register purposes
+			assert_ok!(EncointerReputationCommitments::do_register_purpose(
+				DescriptorType::from_str("Some Description").unwrap()
+			));
+			assert_ok!(EncointerReputationCommitments::do_register_purpose(
+				DescriptorType::from_str("Some Description2").unwrap()
+			));
+
+			// commit
+			assert_ok!(EncointerReputationCommitments::do_commit_reputation(
+				&alice.clone(),
+				cid,
+				1,
+				0,
+				None
+			));
+			assert_ok!(EncointerReputationCommitments::do_commit_reputation(
+				&bob.clone(),
+				cid2,
+				1,
+				0,
+				None
+			));
+			assert_ok!(EncointerReputationCommitments::do_commit_reputation(
+				&alice.clone(),
+				cid,
+				1,
+				1,
+				None
+			));
+			assert_ok!(EncointerReputationCommitments::do_commit_reputation(
+				&bob.clone(),
+				cid,
+				1,
+				1,
+				None
+			));
+
+			assert_ok!(EncointerReputationCommitments::do_commit_reputation(
+				&alice.clone(),
+				cid2,
+				2,
+				0,
+				None
+			));
+			assert_ok!(EncointerReputationCommitments::do_commit_reputation(
+				&bob.clone(),
+				cid,
+				2,
+				0,
+				None
+			));
+			assert_ok!(EncointerReputationCommitments::do_commit_reputation(
+				&alice.clone(),
+				cid,
+				2,
+				1,
+				None
+			));
+			assert_ok!(EncointerReputationCommitments::do_commit_reputation(
+				&bob.clone(),
+				cid2,
+				2,
+				1,
+				None
+			));
+
+			assert!(Commitments::<TestRuntime>::contains_key((cid, 1), (0, &alice)));
+			assert!(Commitments::<TestRuntime>::contains_key((cid2, 1), (0, &bob)));
+			assert!(Commitments::<TestRuntime>::contains_key((cid, 1), (1, &alice)));
+			assert!(Commitments::<TestRuntime>::contains_key((cid, 1), (1, &bob)));
+
+			assert!(Commitments::<TestRuntime>::contains_key((cid2, 2), (0, &alice)));
+			assert!(Commitments::<TestRuntime>::contains_key((cid, 2), (0, &bob)));
+			assert!(Commitments::<TestRuntime>::contains_key((cid, 2), (1, &alice)));
+			assert!(Commitments::<TestRuntime>::contains_key((cid2, 2), (1, &bob)));
+
+			let reputation_lifetime = EncointerCeremonies::reputation_lifetime();
+
+			for _ in 0..reputation_lifetime {
+				run_to_next_phase();
+				run_to_next_phase();
+				run_to_next_phase();
+			}
+
+			assert!(Commitments::<TestRuntime>::contains_key((cid, 1), (0, &alice)));
+			assert!(Commitments::<TestRuntime>::contains_key((cid2, 1), (0, &bob)));
+			assert!(Commitments::<TestRuntime>::contains_key((cid, 1), (1, &alice)));
+			assert!(Commitments::<TestRuntime>::contains_key((cid, 1), (1, &bob)));
+
+			assert!(Commitments::<TestRuntime>::contains_key((cid2, 2), (0, &alice)));
+			assert!(Commitments::<TestRuntime>::contains_key((cid, 2), (0, &bob)));
+			assert!(Commitments::<TestRuntime>::contains_key((cid, 2), (1, &alice)));
+			assert!(Commitments::<TestRuntime>::contains_key((cid2, 2), (1, &bob)));
+
+			run_to_next_phase();
+			run_to_next_phase();
+			run_to_next_phase();
+
+			assert!(!Commitments::<TestRuntime>::contains_key((cid, 1), (0, &alice)));
```

Comment:
hmm, this actually means pruging did not work, right? this test seems to pass although purging failed

Reply:
I believe onCeremonyPhaseChange isn't even linked in the mock, so purging will never happen

Reply:
it is linked in the mock. see [here](https://github.com/encointer/pallets/blob/7752ed427ad7b80437bc161a23a4840f6ae6f982/reputation-commitments/src/mock.rs#L52) and [here](https://github.com/encointer/pallets/blob/7752ed427ad7b80437bc161a23a4840f6ae6f982/test-utils/src/lib.rs#L256)

why would you assume that purging did not work? we assert that the key is not present anymore, so it was purged.